src/log.rs

```diff
@@ -1,94 +1,46 @@
-/// This file establishes a JSON logging mechanism to support `dura stat`.
-use std::fs::{File, OpenOptions, create_dir_all};
-use std::io;
-use std::process;
+use std::fmt::Display;
 
-use serde::{Deserialize, Serialize};
-use chrono::{Utc, Datelike};
-
-use crate::config::Config;
 use crate::snapshots::CaptureStatus;
+use serde::{Deserialize, Serialize};
 
 #[derive(Debug, Serialize, Deserialize)]
 pub enum Operation {
-    Snapshot { repo: String, op: Option<CaptureStatus>, error: Option<String>, latency: f32 },
+    Snapshot {
+        repo: String,
+        op: Option<CaptureStatus>,
+        error: Option<String>,
+        latency: f32,
+    },
 }
 
 impl Operation {
-    fn should_log(&self) -> bool {
-        !matches!(self, Operation::Snapshot { repo: _, op: None, error: None, latency: _ })
-    }
-}
-
-/// Information to log to a log file. These are written in a plain-text structured format, like
-/// JSON.
-#[derive(Debug, Serialize, Deserialize)]
-pub struct Log {
-    time: String,
-    op: Operation,
-}
-
-impl Log {
-    pub fn new(op: Operation) -> Self {
-        Self {
-            time: Utc::now().to_rfc3339(),
-            op,
+    pub fn should_log(&self) -> bool {
+        match self {
+            Operation::Snapshot {
+                repo: _,
+                op,
+                error,
+                latency: _,
+            } => op.is_some() || error.is_some(),
         }
     }
 }
 
-/// For writing log lines out to a file
-pub struct Logger {
-    file: Option<File>,
-}
-
-impl Logger {
-    pub fn new() -> Self {
-        Self { file: None }
-    }
-
-    pub fn write(&mut self, log: Log) {
-        if !log.op.should_log() {
-            return
-        }
-
-        let _ = self.ensure_open();
-        if let Some(ref file) = self.file {
-            let writer = io::LineWriter::new(file);
-            if let Err(error) = serde_json::to_writer(writer, &log) {
-                eprintln!("Unable to write log: {:?}", error);
+impl Display for Operation {
```

Comment:
Oh goodness, you went through all this work to display it, and I'm telling you to do it in JSON, sorry 😊

Reply:
No problem. Should have gone for JSON from the beginning.